code/go/0chain.net/smartcontract/storagesc/blobber_integration_tests.go

```diff
@@ -22,3 +24,30 @@ func afterInsertBlobber(id string) {
 	}
 	return
 }
+
+func afterAddChallenge(challengeID string, validatorsIDs []string) {
+
+}
+
+func beforeEmitAddChallenge(challenge *StorageChallengeResponse) {
+	var (
+		client = crpc.Client()
+		state  = client.State()
+	)
+
+	if state.AdversarialValidator != nil && state.AdversarialValidator.PassAllChallenges && containsString(challenge.ValidatorIDs, state.AdversarialValidator.ID) {
+		// any challenge adulteration produces an invalid challenge
+		challenge.AllocationRoot = strings.ReplaceAll(challenge.AllocationRoot, "1", "0")
+		challenge.AllocationRoot = strings.ReplaceAll(challenge.AllocationRoot, "a", "b")
```

Comment:
if we want to break Allocation root we can shuffle chars probably, this one isnot guaranteed to break it

Reply:
Although it is very unlikely to have the string without an `a` and a `1` it may happen. Changing the implementation to revert the string.